File: decision.java
class Main {
    static int count = 0;

    public static void main(String[] args) {

        for (int i = 0; i < 10; i++) {
            Thread t = new Func();
            t.start();
            try {
                t.join();
            } catch (InterruptedException ex) {}
        }
    }

    public static class Func extends Thread {
        public void run() {
            try {
                sleep((int) Math.random() * 1000 + 1);
            } catch (InterruptedException ex) {}
            System.out.println(count * 2);
            count++;
        }

    }
}

File: decision.js

function rand(min, max){
	return (max-min)*Math.random()+min; //Высчитываем рандомную задержку от 0 до 1с
}

function f(a){
  for (var i = 0 ; i < 10 ; i++){
    console.log(a*i);
  }
}

setInterval(f(2) , rand(0 , 1000)); // Выставляем интервал
